### PR TITLE
fix Issue 22676 - Error: cannot implicitly convert expression 'T!C' of type 'void' to 'string'

### PR DESCRIPTION
**Pull request by ibuclaw**
Opt for lazier running of semantic for class types.  Targeting master because stable branch is closed until the next beta release for 2.099.0

### CODE REVIEW

**Review by thewilsonator — APPROVED** [review]
regressions should go to stable though

File: src/dmd/dclass.d [hunk 1]
```diff
@@ -446,6 +446,7 @@ extern (C++) class ClassDeclaration : AggregateDeclaration
             *poffset = 0;
         while (cd)
         {
+            assert(cd.baseClass || cd.semanticRun >= PASS.semanticdone || cd.isInterfaceDeclaration());
```
Comment (thewilsonator) [inline src/dmd/dclass.d:449]: looks like this doesn't pass (in doc mode?)

**Comment by ibuclaw** [comment]
> regressions should go to stable though

I'd be convinced only if 2.098.2 is going to be tagged.

File: src/dmd/dclass.d (hunk 1 above)
Comment (ibuclaw) [reply src/dmd/dclass.d:449]: This is here to catch other bugs that crop up when semantic on classes are not done eagerly enough.

**Review by RazvanN7 — APPROVED** [review]

File: src/dmd/dclass.d (hunk 1 above)
Comment (ibuclaw) [reply src/dmd/dclass.d:449]: FYI, this was the condition used to call `dsymbolSemantic(cd)` in this function prior to making this pure.  The assert just restores the check to catch any further bad cases.

**Comment by ibuclaw** [comment]
There's at least one more place that needs to be handled. https://issues.dlang.org/show_bug.cgi?id=22714

**Pull request merged**